flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/result/FinalizedCollectStreamResult.java

```diff
@@ -0,0 +1,254 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.client.gateway.local.result;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.streaming.experimental.SocketStreamIterator;
+import org.apache.flink.table.client.SqlClientException;
+import org.apache.flink.table.client.gateway.SqlExecutionException;
+import org.apache.flink.table.client.gateway.local.CollectStreamTableSink;
+import org.apache.flink.table.client.gateway.local.ProgramDeployer;
+import org.apache.flink.table.sinks.TableSink;
+import org.apache.flink.types.Row;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Signal;
+import sun.misc.SignalHandler;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Collects results and returns them as finalized table.
+ *
+ * @param <C> cluster id to which this result belongs to
+ */
+public class FinalizedCollectStreamResult<C> extends BasicResult<C> implements DynamicResult<C>, FinalizedResult<C> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(FinalizedCollectStreamResult.class);
+	private final TypeInformation<Row> outputType;
+	private final SocketStreamIterator<Tuple2<Boolean, Row>> iterator;
+	private final CollectStreamTableSink collectTableSink;
+	private final ResultRetrievalThread retrievalThread;
+	private final CountDownLatch resultLatch;
+	private final List<Row> finalizedTable;
+	/**
+	 * Caches the last row position for faster access. The position might not be exact (if rows
```

Comment:
You copied this from an other class, right? Please abstract correctly to remove code duplication.

Reply:
Yes, copied some code from CollectStreamResult. Will do a retractor to remove duplication code.